app/main.py

```diff
@@ -2,0 +1,51 @@
+class SoftwareEngineer:
+    def __init__(self, name: str) -> None :
+        self.name = name
+        self.skills = []
+
+    def learn_skill(self, skill: str) -> None :
+        self.skills.append(skill)
+
+
+class FrontendDeveloper(SoftwareEngineer):
+    def __init__(self, name: str):
+        super().__init__(name)
+        for skill in ["JavaScript", "HTML", "CSS"]:
+            self.learn_skill(skill)
+
+    def create_awesome_web_page(self) -> str:
+        print(f"{self.name} is creating a webpage...")
+        return "<h1>Hello world</h1>"
+
+
+class BackendDeveloper(FrontendDeveloper):
+    def __init__(self, name: str) -> None :
+        super().__init__(name)
+        # self.skills.clear()
+        for skill in ["Python", "SQL", "Django"]:
+            self.learn_skill(skill)
+
+    def create_powerful_api(self) -> str :
+        print(f"{self.name} is creating an API...")
+        return "http://127.0.0.1:8000"
+
+
+class AndroidDeveloper(SoftwareEngineer):
+    def __init__(self, name: str) -> None :
+        super().__init__(name)
+        for skill in ["Java", "Android studio"]:
+            self.learn_skill(skill)
+
+    def create_smooth_mobile_app(self) -> str:
+        print(f"{self.name} is creating a mobile app...")
+        return "Ads every three swipes"
+
+
+class FullStackDeveloper(BackendDeveloper, FrontendDeveloper):
```

Comment:
The `FullStackDeveloper` class inherits from both `BackendDeveloper` and `FrontendDeveloper`. However, due to Python's method resolution order (MRO), the `BackendDeveloper` class will be initialized first, which might not be the intended behavior. Ensure that the class hierarchy and MRO align with the desired functionality.